### DISCUSSION TITLE
NFT log group

### DISCUSSION THREAD

**Discussion by volga629-1** [FoobarOy/foomuuri#33] [General]
Hello Everyone, 
I added the following rule 

```
fw-wan1 {
        template out_fw_services nft "log prefix Accept-fw2wan group 1"
        # global deny rule
        reject nft "log prefix GlobReject-fw2wan group 2"
}

```

Which produce nft list ruleset 

```
	chain fw-wan1_4 {
		icmp type echo-request accept
		jump allow_icmp_4
		ct state vmap { invalid : goto invalid_drop, established : accept, related : accept, new : jump smurfs_4, untracked : jump smurfs_4 }
		meta pkttype { broadcast, multicast } drop
		udp dport 443 accept
		tcp dport 443 accept
		log prefix "GlobReject-fw2wan" group 2
	}

```

As you see log entry is missing, do I need add on  template service definition ?

**Comment by kimheino** [comment]
Foomuuri supports nflog without `nft` rules.

>         template out_fw_services nft "log prefix Accept-fw2wan group 1"

You can not add nft or other matchers/rules to template call. You must implement logging inside template:

```
template out_fw_services {
  https  log "Accept-fw2wan" log_level "group 1"
}

fw-wan1 {
  template out_fw_services
```

>         # global deny rule
>         reject nft "log prefix GlobReject-fw2wan group 2"

That must be:

```
  reject log "GlobReject-fw2wan" log_level "group 2"
```

Or two separate rules:

```
  nft "log prefix GlobReject-fw2wan group 2"
  reject
```

**Comment by volga629-1** [comment]
Thank you for detail explanation.
